### PR TITLE
Restructure and improve docs to highlight connecting via EIP-6963

### LINKED ISSUE #1197: Improve connecting via EIP-6963 content
Now that EIP-6963 is the recommended method of connecting to MetaMask, and we also want to [deprioritize the SDK content](https://github.com/MetaMask/metamask-docs/issues/1196), some of our documentation structure and code samples are outdated. This issue is to:

- Remove the [Quickstart](https://docs.metamask.io/wallet/quickstart/) section
- Update the [Detect wallet](https://docs.metamask.io/wallet/how-to/detect-wallet/) section:
  - Rename it to "Connect to MetaMask"
  - Recommend EIP-6963 as the first option for integrating, whether directly or via third-party library
  - Add EIP-6963 code examples for vanilla JS
  - We no longer recommend [detecting MetaMask](https://docs.metamask.io/wallet/how-to/detect-wallet/metamask/) via `@metamask/detect-provider` (should this content be removed completely?)
- Audit and update code samples throughout the docs that use `window.ethereum`
- Remove [Access a user's MetaMask provider](https://docs.metamask.io/wallet/how-to/access-provider/)

Other places that may require related updates:
- Ethereum provider API [concept](https://docs.metamask.io/wallet/concepts/wallet-api/#ethereum-provider-api) and [reference](https://docs.metamask.io/wallet/reference/provider-api/)
- [Wallet interoperability](https://docs.metamask.io/wallet/concepts/wallet-interoperabilty/)
- [Detect wallet](https://docs.metamask.io/snaps/how-to/connect-to-a-snap/#detect-wallet) when connecting to a Snap

### PR DESCRIPTION
**Pull request by alexandratran**
This PR addresses #1197. Updates include:

- Remove quickstarts, rearrange some how-to sections, and add a "Connect to MetaMask" section that highlights connecting via EIP-6963. Preview: https://docs.metamask.io/1197-improve-eip-6963/wallet/how-to/connect/
- Include admonitions on relevant pages to highlight the alternative discovery mechanism. Preview:
  - [Detect MetaMask](https://docs.metamask.io/1197-improve-eip-6963/wallet/how-to/connect/detect-metamask/)
  - [Ethereum provider API concept](https://docs.metamask.io/1197-improve-eip-6963/wallet/concepts/wallet-api/#ethereum-provider-api)
  - [Ethereum provider API reference](https://docs.metamask.io/1197-improve-eip-6963/wallet/reference/provider-api/)
- Update code samples throughout the docs, including the provider API reference, to use `provider` instead of `window.ethereum`. All instances include an in-line comment mentioning to use `window.ethereum` if you don't support EIP-6963. For example: https://docs.metamask.io/1197-improve-eip-6963/wallet/how-to/manage-networks/detect-network/
- Other minor related updates.

To do (for future PRs):

- #1215
- #1216

### CODE REVIEW

**Review by Montoya — APPROVED** [review]
Massive improvement. We should consider addressing https://docs.metamask.io/wallet/concepts/convenience-libraries/ in a separate ticket, but this is great for now.

**Review by Montoya — APPROVED** [review]

**Review by bgravenorst — COMMENTED** [review]
Added some feedback.

File: wallet/how-to/connect/index.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+# Connect to MetaMask
+
+You can connect to your users' MetaMask wallets from your dapp by detecting MetaMask in their
+browsers and connecting to their accounts.
+
+This page describes how to connect to MetaMask using the alternative wallet detection mechanism
+introduced by [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963).
+EIP-6963 enables
+[wallet interoperability](../../concepts/wallet-interoperabilty.md), and shifts dapps from
+relying solely on [`window.ethereum`](detect-metamask.md) for wallet detection.
```
[the remark this answers is not in the document]
Comment (alexandratran) [reply wallet/how-to/connect/index.md:15]: Removing "alternative" but keeping the two sentences as the description is quite long, and to clearly identify "EIP-6963" as the subject of sentence two.

File: wallet/how-to/connect/index.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+## Use third-party libraries
+
+You can connect to MetaMask using the following third-party libraries that support EIP-6963:
+
+- [Wagmi 2+](https://wagmi.sh)
+- [Web3Modal 3+](https://docs.walletconnect.com/web3modal/about)
+- [MIPD Store](https://github.com/wevm/mipd)
+- [RainbowKit](https://www.rainbowkit.com/)
+
+## Connect to MetaMask directly
```
Comment (bgravenorst) [inline wallet/how-to/connect/index.md:30]: Should this be something like, "Connect using the Wallet API"?

File: wallet/how-to/connect/detect-metamask.md [hunk 3]
```diff
@@ -5,6 +5,16 @@ sidebar_position: 1
 
 # Detect MetaMask
 
+:::caution Important
+MetaMask supports [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963), which introduces an
+alternative wallet detection mechanism to the `window.ethereum` injected provider.
+This alternative mechanism enables dapps to support [wallet interoperability](../../concepts/wallet-interoperability.md)
+by discovering multiple injected wallet providers in a user's browser.
+
+This page describes the legacy method of connecting to MetaMask using `window.ethereum`.
```
Comment (bgravenorst) [inline wallet/how-to/connect/detect-metamask.md:14]: Can we start with this sentence instead?

File: wallet/how-to/connect/detect-metamask.md [hunk 4]
```diff
@@ -5,6 +5,16 @@ sidebar_position: 1
 
 # Detect MetaMask
```
Comment (bgravenorst) [inline wallet/how-to/connect/detect-metamask.md:6]: Should we still document this if we no longer recommend users connect using this legacy method?

**Review by bgravenorst — APPROVED** [review]
Some additional feedback.

File: wallet/how-to/connect/detect-metamask.md (hunk 4 above)
Comment (alexandratran) [reply wallet/how-to/connect/detect-metamask.md:6]: The tutorials still use this method and link to this page. Those will updated in a future PR. We also still want to document this method for a beginner use case – may move this to a tutorial in the future.

File: wallet/how-to/connect/index.md (hunk 2 above)
Comment (alexandratran) [reply wallet/how-to/connect/index.md:30]: The Wallet API isn't explicitly mentioned in most of our other how-tos, so I hesitate to add it to the header here, for consistency. We're really just contrasting that you're implementing it directly in your dapp code vs. via 3rd-party.

**Pull request merged**